src/time-picker/TimePicker.js

```diff
@@ -155,6 +156,8 @@ export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
   return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
 };
 
+export const sortTimeList = timeList => sortBy(timeList, time => ( time.hours === 0 ? 24 : time.hours));
```

Comment:
But this is concerning another issue is not directly related to this one, and is a design one. (that is actually scary, and hopefully you found it)